cloudinit/sources/DataSourceWSL.py

```diff
@@ -143,22 +143,35 @@ def candidate_user_data_file_names(instance_name) -> List[str]:
     ]
 
 
-def load_yaml_or_bin(data_path: str) -> Optional[Union[dict, bytes]]:
-    """
-    Tries to load a YAML file as a dict, otherwise returns the file's raw
-    binary contents as `bytes`. Returns `None` if no file is found.
-    """
-    try:
-        bin_data = util.load_binary_file(data_path)
-        dict_data = util.load_yaml(bin_data)
-        if dict_data is None:
-            return bin_data
+class ConfigData:
+    """Models a piece of configuration data as a dict if possible, while
+    retaining its raw representation alongside its file path"""
 
-        return dict_data
-    except FileNotFoundError:
-        LOG.debug("No data found at %s, ignoring.", data_path)
+    def __init__(self, path: PurePath):
+        if not os.path.isfile(path.as_posix()):
+            raise FileNotFoundError(f"Config file {path} not found")
+
+        self._raw_data: str = util.load_binary_file(path).decode("utf-8")
+        self._raw_data = self._raw_data.lstrip()
```

Comment:
Why are we lstriping raw_data? Shouldn't we be passing this around unaltered and undecoded?

Reply:
lstrip was for later checking if it started with `#cloud-config` stenza, which is no longer necessary as we are now comparing the MIME type with the output of `cloudinit.handlers.type_from_starts_with()`.